src/main/java/edu/yu/cs/artAPI/ArtResource.java

```diff
@@ -1,47 +1,101 @@
 package edu.yu.cs.artAPI;
 
+import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
+import static javax.ws.rs.core.Response.Status.NOT_FOUND;
+
 import java.util.List;
 
+import javax.inject.Inject;
 import javax.transaction.Transactional;
 import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
 import javax.ws.rs.GET;
+import javax.ws.rs.NotFoundException;
 import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
 import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
 import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
 import javax.ws.rs.core.Response.Status;
 
-@Path("/art")
+import edu.yu.cs.artAPI.repositories.ArtRepository;
+import edu.yu.cs.artAPI.repositories.GalleryRepository;
+
+@Path("/galleries")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
 public class ArtResource {
+
+    @Inject ArtRepository ar;
+    @Inject GalleryRepository gr; 
+    
+         
+    @GET
+    @Path("/{gallery-id}/arts")
+    //Gets all of the art from a specific gallery
+    public List<Art> getAll(@PathParam("gallery-id") long galleryId) {
+        return gr.findByIdOptional(galleryId).orElseThrow(NotFoundException::new).artList;
+    }
     
     @GET
-    @Produces(MediaType.APPLICATION_JSON)
-    public List<Art> allArt() {
-        return Art.listAll();
+    @Path("/{gallery-id}/arts/namesearch")
```

Comment:
IMO, this could use the same path as `//Gets all of the art from a specific gallery` - I would argue that getting all the arts is a specific use case of the more general `Get specific arts from a gallery`. 

The `name` could be an optional query parameter - if its absent, then get all. Otherwise, get the arts specified by the search criteria.